specification/digitaltwins/resource-manager/Microsoft.DigitalTwins/stable/2023-01-31/digitaltwins.json

```diff
@@ -0,0 +1,2328 @@
+            "Moving",
+            "Disabled"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "TimeSeriesDatabaseConnectionState",
+            "modelAsString": true
+          }
+        },
+        "identity": {
+          "$ref": "#/definitions/ManagedIdentityReference",
+          "description": "Managed identity properties for the time series database connection resource.",
+          "x-nullable": true
+        }
+      }
+    },
+    "TimeSeriesDatabaseConnection": {
+      "description": "Describes a time series database connection resource.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ExternalResource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Properties of a specific time series database connection.",
+          "$ref": "#/definitions/TimeSeriesDatabaseConnectionProperties"
+        }
+      }
+    },
+    "AzureDataExplorerConnectionProperties": {
+      "description": "Properties of a time series database connection to Azure Data Explorer with data being sent via an EventHub.",
+      "x-ms-discriminator-value": "AzureDataExplorer",
+      "required": [
+        "adxEndpointUri",
+        "adxResourceId",
+        "adxDatabaseName",
+        "eventHubEndpointUri",
+        "eventHubEntityPath",
+        "eventHubNamespaceResourceId"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/TimeSeriesDatabaseConnectionProperties"
+        }
+      ],
+      "properties": {
+        "adxResourceId": {
+          "description": "The resource ID of the Azure Data Explorer cluster.",
+          "type": "string"
+        },
+        "adxEndpointUri": {
+          "description": "The URI of the Azure Data Explorer endpoint.",
+          "type": "string"
+        },
+        "adxDatabaseName": {
+          "description": "The name of the Azure Data Explorer database.",
+          "type": "string"
+        },
+        "adxTableName": {
+          "description": "The name of the Azure Data Explorer table used for storing updates to properties of twins and relationships. Defaults to AdtPropertyEvents.",
+          "type": "string",
+          "default": "AdtPropertyEvents",
+          "x-nullable": true
+        },
+        "adxTwinLifecycleEventsTableName": {
+          "description": "The name of the Azure Data Explorer table used for recording twin lifecycle events. The table will not be created if this property is left unspecified.",
+          "type": "string",
+          "x-nullable": true
+        },
+        "adxRelationshipLifecycleEventsTableName": {
+          "description": "The name of the Azure Data Explorer table used for recording relationship lifecycle events. The table will not be created if this property is left unspecified.",
+          "type": "string",
+          "x-nullable": true
+        },
```

Comment:
Consider making this an eunum : 


Explained here : [https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices](https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices)


Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].

Reply:
This is not a blocking comment , so I am signing off , please fix before checking in